src/Polysemy/Internal/Tactics.hs

```diff
@@ -38,41 +45,53 @@ import Polysemy.Internal.Union
 -- data 'Polysemy.Resource.Resource' m a where
 --   'Polysemy.Resource.Bracket' :: m a -> (a -> m ()) -> (a -> m b) -> 'Polysemy.Resource.Resource' m b
 -- @
---
--- Here we have an @m a@ which clearly needs to be run first, and then
--- subsequently call the @a -> m ()@ and @a -> m b@ arguments. In a 'Tactical'
--- environment, we can write the threading code thusly:
+-- We can interpret this effect like so:
 --
 -- @
--- 'Polysemy.Resource.Bracket' alloc dealloc use -> do
---   alloc'   <- 'runT'  alloc
---   dealloc' <- 'bindT' dealloc
---   use'     <- 'bindT' use
+-- runResource = Polysemy.interpretH $ \\case
+--  'Polysemy.Resource.Bracket' alloc dealloc use -> do
+--    resource <- 'runTSimple' alloc
+--    result <- 'bindTSimple' use resource
+--    'bindTSimple' dealloc resource
```

Comment:
This is wrong, since this will have `dealloc` inherit the state from `alloc` and not `result`.
The "proper" would be:
```haskell
  fresource <- runTSimple alloc
  fresourceresult <- bindTSimple (\a -> fmap ((,) a) <$> use a) fresource
  bindTSimple (\(resource, result) ->  result <$ dealloc resource) fresourceresult
```
This is very complicated. Also, this documentation is misleading since this isn't actually how `runResource` is implemented; it does some extra stuff like using the inspector to check if `use` failed.

All in all, we should pick a different example than `Resource`.

Reply:
Hm, does the current example do the same thing regarding the "state inheritance"? This is the impression I was left with (and it also seems to be the point of the example), from the current documentation.

I guess it would be misleading if the actual `runResource` runs in a different way.